src/test/scala/chiselTests/QueueFlushSpec.scala

```diff
@@ -0,0 +1,92 @@
+package chiselTests
+
+import org.scalacheck._
+
+import chisel3._
+import chisel3.testers.BasicTester
+import chisel3.util._
+import chisel3.util.random.LFSR
+
+class ThingsPassThroughFlushQueueTester(elements: Seq[Int], queueDepth: Int, bitWidth: Int, tap: Int, useSyncReadMem: Boolean) extends BasicTester {
+  
+  val q = Module(new Queue(UInt(bitWidth.W), queueDepth, hasFlush = true))  
+  q.io.flush.get := false.B
+  val elems = VecInit(elements.map {
+    _.asUInt()
+  })
```

Comment:
```suggestion
  val elems = VecInit(elements.map(_.U))
```
We generally use `.U` for `Int => UInt`, and `.asUInt` for `Data => UInt`